cmd_unix.go

```diff
@@ -0,0 +1,39 @@
+// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
+
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"syscall"
+)
+
+// Default location of hosts-file on Unix.
+var hostsFile = "/etc/hosts"
+
+// tempFile creates a new temporary hosts-file in an appropriate directory,
+// opens the file for writing, and returns the resulting *os.File.
+func tempFile() (*os.File, error) {
+	fs, err := os.Stat(hostsFile)
```

Comment:
I'm a little annoyed that we have to open /etc/hosts once (as `os.Open(*fileArg)`) and then again here. Can we just do it one time? Also maybe on Windows we should open it in the same directory as whatever file argument is passed? It's not always the hosts file.

Reply:
Good point. Perhaps change to something like: `func tempFile(hostsFile *os.File) (*os.File, error)`? That should solve both issues (using `hostsFile.Stat()` and `hostsFile.Name()`).

Reply:
Yeah that sounds good to me.

Reply:
I see that the tool supports reading hosts-files via stdin. This means we can't always rely on a file having already been opened. Perhaps it's cleaner to just provide the path as an argument, even if it means a file is usually "opened" twice, to avoid the extra boilerplate?

Reply:
Good point. Sounds fine with me

(and apologies for the time difference... I'm on the west coast of the USA)

Reply:
No problem (I'm in Norway).